### ISSUE TITLE
Coding Standards and Style

### ISSUE THREAD

**Issue by amadio** [apt-sim/AdePT#73]
Let's use this issue to discuss about and decide on a set of coding style, standards, and other guidelines.

A new project offers us an opportunity to adopt more modern coding style than what is used by ROOT/Geant4.
I suggest we look into existing widely used standards (including ROOT/Geant4), and pick something based on them to follow.

**References**:
- [Geant4 Coding Guidelines](https://geant4.web.cern.ch/collaboration/coding_guidelines)
- [GNU Coding Standards](https://www.gnu.org/prep/standards/)
- [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
- [Indentation Style (Wikipedia)](https://en.wikipedia.org/wiki/Indentation_style)
- [ISO C++ Coding Standards](https://isocpp.org/wiki/faq/coding-standards)
- [Linux Kernel Coding Style](https://www.kernel.org/doc/html/v4.10/process/coding-style.html)
- [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html)
- [ROOT Coding Conventions](https://root.cern/contribute/coding_conventions)
- [WebKit Code Style Guidelines](https://webkit.org/code-style-guidelines/)

**Comment by hahnjo** [comment]
I think the style question is already done, we have a `.clang-format` (see #10 and #12). Is there anything in particular about the semantic aspect (like naming of attributes) that you'd like to discuss?

**Comment by amadio** [comment]
Yes, clang-format only tells you how to format your code, and I agree that having a tool do it and just accept its output as the right way to format is best. However that does not cover how to naming, and many other things from the links above that are a matter of style.

**Comment by bernhardmgruber** [comment]
There are further options in clang-tidy. For naming you have e.g.: https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html

**Comment by amadio** [comment]
Thanks, @bernhardmgruber. Looks like a lot of work to configure, but should do the job well once that's done.
